### PR TITLE
dialects: (stablehlo) add stablehlo.precision

### PR DESCRIPTION
**Pull request by superlopuh**
One of the attributes necessary to express dot products

CC @efferifick

### CODE REVIEW

File: xdsl/dialects/stablehlo.py [hunk 1]
```diff
@@ -43,6 +51,32 @@ def __init__(
...
+
+# region Attributes
+
+
+class Precision(StrEnum):
+    """
+    XLA precision for an operand. Has backend specific meaning.
+    """
+
+    DEFAULT = "DEFAULT"
```
Comment (alexarice) [inline xdsl/dialects/stablehlo.py:64]: I believe you could use `auto()` here

**Review by alexarice — APPROVED** [review]

File: xdsl/dialects/stablehlo.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/dialects/stablehlo.py:64]: For some reason it made the string lowercase and my test failed

Comment (superlopuh) [reply xdsl/dialects/stablehlo.py:64]: `# |                                    Expected `default`, `HIGH`, or `HIGHEST`.`

Comment (alexarice) [reply xdsl/dialects/stablehlo.py:64]: Yep, documentation seems to say that `auto` lowercases

**Pull request merged**